content/en/contribute/code/static-analysis.md

```diff
@@ -0,0 +1,128 @@
+
+### PR Analysis
+
+SonarCloud is integrated with the CHT GitHub repositories and runs on every pull request. The results are posted as a comment on the pull request. If the Sonar analysis fails the quality check, the pull request will be blocked from merging.
+
+#### What should I do if Sonar finds an issue?
+
+When Sonar flags an issue with the code in your Pull Request, use this decision tree to determine the proper mitigation:
+
+1. If the issue is a genuine concern that should be addressed:
+   1. Fix it and push the updated code to your PR. The PR will automatically be unblocked once the Sonar analysis succeeds.  
+1. If the issue is a "false-positive" (i.e. Sonar has flagged some particular code as violating a rule, but it does not make sense to apply the rule in that context):
+   1. If the rule is one that should not be applied to any CHT code:
+      1. [Remove the rule]({{< ref "#removing-a-rule" >}}) from the default Quality Profile.
+   1. If it does not make sense to apply the rule to this particular code, you can do one of the following:
+      1. Completely ignore Sonar issues [on that line of code](https://docs.sonarsource.com/sonarqube/latest/user-guide/issues/#technical-review) by adding the `// NOSONAR` comment to the end of the line.
+      1. Update the `.sonarcloud.properties` to [ignore _that rule_ for that particular file]({{< ref "#ignoring-a-specific-rule-for-a-file" >}}).
+      1. Update the `.sonarcloud.properties` to [ignore _all rules_ for that particular file]({{< ref "#ignoring-all-rules-for-a-file" >}}) (useful if the file has been copied from an external dependency).
+
+## Adding a new repo to SonarCloud
+
+Add a `.sonarcloud.properties` file to the repository with desired your [repo-level configuration]({{< ref "#ignoring-all-rules-for-a-file" >}}).
+
+In the GitHub UI, navigate to the settings for the `medic` org > Integrations > Applications > SonarCloud and click the `Configure` button.
+
+In the Repository access section, select your desired repository from the drop-down and click `Save`.
+
+You will be automatically redirected to the SonarCloud UI where you can configure the repo-level settings. Use the `+` button and choose `Analyze new project`.  Select the repo from the list and click `Set Up`.
+
+### New Code Definition
+
+When setting up a new repository in SonarCloud, you will be asked to define what is considered to be "new code". This is used to determine which code in the default branch is considered "new" (affects reporting of issues, etc).
+
+Consult [the documentation](https://docs.sonarcloud.io/improving/new-code-definition/) for more details on the options available. For projects that do not use Gradle or Maven for version management, the `Number of days` option is recommended (since `Previous version` would require maintaining a version number in the `.sonarcloud.properties` file).
```

Comment:
How will the NCD be used? If I'm submitting a PR will it block if someone else changed that same file within the NCD period? Or is it only when analysing the default branch? If so why not also analyse the old code?

I guess what I'm asking is, what is the workflow which is improved by the distinction of new vs old code?

Reply:
As far as I can tell, if we use the "Automatic" analysis method and require that all PRs pass the Quality Gate, then the NCD is not really very important to us.  In general, I think the point of an NCD was as a generic way to incrementally measure code quality (that did not rely on specific implementation details of an SCM like git/GitHub).  If your code creation workflow does not involve branches/PRs/etc, then you need some way of deciding what should be analyzed as "changed" when trying to determine if the code base is in an acceptable state.  

Like I said, though, I don't think this really matters much to us since, with a PR, the NCD is not applied and just code code changed in the PR is considered "new".  The NCD will get used when Sonar analyzes the `master` branch, but since all our PRs should already be passing the Quality Gate, then it should not be possible for there to be any issues in "new" code in the `master` branch.